packages/main/src/TextArea.ts

```diff
@@ -0,0 +1,706 @@
+	 * @public
+		 */
+	@property()
+	value!: string;
+	/**
+	 * Indicates whether the user can interact with the component or not.
+	 * <br><br>
+	 * <b>Note:</b> A disabled component is completely noninteractive.
+	 *
+	 * @type {boolean}
+	 * @name sap.ui.webc.main.TextArea.prototype.disabled
+	 * @defaultvalue false
+	 * @public
+	 */
+	@property({ type: Boolean })
+	disabled!: boolean;
+	/**
+	 * Defines whether the component is read-only.
+	 * <br><br>
+	 * <b>Note:</b> A read-only component is not editable,
+	 * but still provides visual feedback upon user interaction.
+	 *
+	 * @type {boolean}
+	 * @name sap.ui.webc.main.TextArea.prototype.readonly
+	 * @defaultvalue false
+	 * @public
+	 */
+	@property({ type: Boolean })
+	readonly!: boolean;
+	/**
+	 * Defines whether the component is required.
+	 *
+	 * @type {boolean}
+	 * @name sap.ui.webc.main.TextArea.prototype.required
+	 * @defaultvalue false
+	 * @public
+	 * @since 1.0.0-rc.3
+	 */
+	@property({ type: Boolean })
+	required!: boolean;
+
+	/**
+	 * Defines a short hint intended to aid the user with data entry when the component has no value.
+	 *
+	 * @type {string}
+	 * @name sap.ui.webc.main.TextArea.prototype.placeholder
+	 * @defaultvalue ""
+	 * @public
+	 */
+	@property()
+	placeholder!: string;
+
+	/**
+	 * Defines the value state of the component.
+	 * <br><br>
+	 * Available options are:
+	 * <ul>
+	 * <li><code>None</code></li>
+	 * <li><code>Error</code></li>
+	 * <li><code>Warning</code></li>
+	 * <li><code>Success</code></li>
+	 * <li><code>Information</code></li>
+	 * </ul>
+	 * <br><br>
+	 * <b>Note:</b> If <code>maxlength</code> property is set,
+	 * the component turns into "Warning" state once the characters exceeds the limit.
+	 * In this case, only the "Error" state is considered and can be applied.
+	 * @type {sap.ui.webc.base.types.ValueState}
+	 * @name sap.ui.webc.main.TextArea.prototype.valueState
+	 * @defaultvalue "None"
+	 * @since 1.0.0-rc.7
+	 * @public
+	 */
+	@property({ type: ValueState, defaultValue: ValueState.None })
+	valueState!: ValueState;
+
+	/**
+	 * Defines the number of visible text lines for the component.
+	 * <br><br>
+	 * <b>Notes:</b>
+	 * <ul>
+	 * <li>If the <code>growing</code> property is enabled, this property defines the minimum rows to be displayed
+	 * in the textarea.</li>
+	 * <li>The CSS <code>height</code> property wins over the <code>rows</code> property, if both are set.</li>
+	 * </ul>
+	 *
+	 * @type {sap.ui.webc.base.types.Integer}
+	 * @name sap.ui.webc.main.TextArea.prototype.rows
+	 * @defaultvalue 0
+	 * @public
+	 */
+	@property({ validator: Integer, defaultValue: 0 })
+	rows!: number;
+
+	/**
+	 * Defines the maximum number of characters that the <code>value</code> can have.
+	 *
+	 * @type {sap.ui.webc.base.types.Integer}
+	 * @name sap.ui.webc.main.TextArea.prototype.maxLength
+	 * @defaultValue null
+	 * @public
+	 */
+	@property({ validator: Integer, defaultValue: null })
+	maxlength!: number;
```

Comment:
If the default value is null and maxLength could be null, maxlength is not guaranteed as the !: syntax suggests. If really the maxlength could be null, then "?:" is more accurate.